src/lib.rs

```diff
@@ -534,76 +534,76 @@ impl Plugin for EguiPlugin {
         world.init_resource::<EguiUserTextures>();
         world.init_resource::<EguiMousePosition>();
 
-        app.add_startup_systems(
+        app.add_systems(PreStartup,
             (
                 setup_new_windows_system,
-                apply_system_buffers,
+                apply_deferred,
                 update_window_contexts_system,
             )
                 .chain()
-                .in_set(EguiStartupSet::InitContexts)
-                .in_base_set(StartupSet::PreStartup),
+                .in_set(EguiStartupSet::InitContexts),
         );
-        app.add_systems(
+        app.add_systems(PreUpdate,
             (
                 setup_new_windows_system,
-                apply_system_buffers,
+                apply_deferred,
                 update_window_contexts_system,
             )
                 .chain()
-                .in_set(EguiSet::InitContexts)
-                .in_base_set(CoreSet::PreUpdate),
+                .in_set(EguiSet::InitContexts),
         );
-        app.add_system(
+        app.add_systems(PreUpdate, 
             process_input_system
                 .in_set(EguiSet::ProcessInput)
                 .after(InputSystem)
-                .after(EguiSet::InitContexts)
-                .in_base_set(CoreSet::PreUpdate),
+                .after(EguiSet::InitContexts),
         );
-        app.add_system(
+        app.add_systems(PreUpdate,
             begin_frame_system
                 .in_set(EguiSet::BeginFrame)
-                .after(EguiSet::ProcessInput)
-                .in_base_set(CoreSet::PreUpdate),
+                .after(EguiSet::ProcessInput),
         );
-        app.add_system(
+        app.add_systems(PostUpdate,
             process_output_system
-                .in_set(EguiSet::ProcessOutput)
-                .in_base_set(CoreSet::PostUpdate),
+                .in_set(EguiSet::ProcessOutput),
         );
-        app.add_system(
+        app.add_systems(PostUpdate,
             update_egui_textures_system
-                .after(EguiSet::ProcessOutput)
-                .in_base_set(CoreSet::PostUpdate),
+                .after(EguiSet::ProcessOutput),
         );
-        app.add_system(free_egui_textures_system.in_base_set(CoreSet::Last));
+        app.add_systems(Last, free_egui_textures_system)
+                .add_systems(Render,
+                    render_systems::prepare_egui_transforms_system.in_set(RenderSet::Prepare),
+                )
+                .add_systems(Render, render_systems::queue_bind_groups_system.in_set(RenderSet::Queue))
+                .add_systems(Render, render_systems::queue_pipelines_system.in_set(RenderSet::Queue));
 
         let mut shaders = app.world.resource_mut::<Assets<Shader>>();
         shaders.set_untracked(
             EGUI_SHADER_HANDLE,
             Shader::from_wgsl(include_str!("egui.wgsl")),
         );
+    }
 
+    fn finish(&self, app: &mut App) {
```

Comment:
not sure if necessary, I did that because I had a crash due to wrongly using `Update` instead of `Render` ; and noticed the post processing example did initialize its resources in `finish` rather than `build` ; probably a better practice to allow other plugins to prewrite the resource ? not sure though, please challenge :)

Reply:
Sounds good to me